core/common/src/main/java/alluxio/conf/AlluxioProperties.java

```diff
@@ -191,7 +191,8 @@ public boolean isSet(PropertyKey key) {
   public boolean isSetByUser(PropertyKey key) {
     if (mUserProps.containsKey(key)) {
       Optional<String> val = mUserProps.get(key);
-      return val.isPresent();
+      // Sources larger than Source.CLUSTER_DEFAULT are considered to be set by the user
+      return val.isPresent() && (getSource(key).compareTo(Source.CLUSTER_DEFAULT) > 0);
```

Comment:
even `CLUSTER_DEFAULT` can be set by users (admin).
maybe just check if it is not equal to `UNKNOWN` or `DEFAULT`?

Reply:
In the picture, I did not manually set `UNDERFS_VERSION`,
After the Worker merges the cluster configuration, `getSource(PropertyKey.UNDERFS_VERSION)` returns to `CLUSTER_DEFAULT`.
If isSetByUser()  just check if it is not equal to `UNKNOWN` or `DEFAULT` here `isSetByUser()` will return `true` which is not accurate

Why even `CLUSTER_DEFAULT` can be set by users? This does not match the literal meaning of `CLUSTER_DEFAULT`, is it possible to add another `Source.Type` to distinguish it better?

![image](https://user-images.githubusercontent.com/32928346/150632205-f85a1f1a-88bc-457e-8065-afab3d6755c8.png)